src/functions/utils.php

```diff
@@ -595,138 +602,343 @@ function has_blocks( $post = null ) {
 
 		/**
 		 * Allow plugins to customize REST API arguments and callbacks.
 		 *
-		 * @param array  $args      Either an array of options for the endpoint, or an array of arrays for
-		 *                          multiple methods. Default empty array.
+		 * @since 4.9.12
+		 *
 		 * @param string $namespace The first URL segment after core prefix. Should be unique to your package/plugin.
 		 * @param string $route     The base URL for route you are adding.
 		 * @param bool   $override  Optional. If the route already exists, should we override it? True overrides,
 		 *                          false merges (with newer overriding if duplicate keys exist). Default false.
 		 *
-		 * @since 4.9.12
+		 * @param array  $args      Either an array of options for the endpoint, or an array of arrays for
+		 *                          multiple methods. Default empty array.
 		 */
 		$args = apply_filters( 'tribe_register_rest_route_args', $args, $namespace, $route, $override );
+
 		return register_rest_route( $namespace, $route, $args, $override );
 	}
 }
 
-/**
- * Gets the earliest version we have recorded for the plugin.
- * Defaults to the current version if there are no records.
- * If no version info found, it will return false.
- *
- * @since TBD
- *
- * @param object $class The plugin object.
- *
- * @return string|boolean The SemVer version string, or false if no info found.
- */
-function tribe_get_install_version( $class ) {
-	// Try for the version history first.
-	if ( ! empty( $class->version_history_slug ) ) {
-		return Tribe__Settings_Manager::get_option( $class->version_history_slug );
-	}
+if ( ! function_exists( 'tribe_get_class_instance' ) ) {
+	/**
+	 * Gets the class instance / Tribe Container from the passed object or string.
+	 *
+	 * @since TBD
+	 *
+	 * @see   \tad_DI52_Container::isBound()
+	 * @see   \tribe()
+	 *
+	 * @param string|object $class The plugin class' singleton name, class name, or instance.
+	 *
+	 * @return mixed|object|Tribe__Container|null
+	 */
+	function tribe_get_class_instance( $class ) {
+		$instance = null;
+
+		if ( is_object( $class ) ) {
+			$instance = $class;
+		} elseif ( is_string( $class ) ) {
+			// Check if class exists and has instance getter method.
+			if ( class_exists( $class ) ) {
+				if ( method_exists( $class, 'instance' ) ) {
+					$instance = $class::instance();
+				} elseif ( method_exists( $class, 'get_instance' ) ) {
+					$instance = $class::get_instance();
+				}
+			}
 
-	// Fall back to the current plugin version.
-	if ( ! empty( $class::VERSION ) ) {
-		return $class::VERSION;
-	}
+			// Check if we can run tribe() on the singleton string.
+			if ( ! $instance ) {
+				foreach ( $GLOBALS as $global_item ) {
+					if ( $global_item instanceof Tribe__Container ) {
+						if ( $global_item->isBound( $class ) ) {
+							$instance = tribe( $class );
+						}
+
+						break;
+					}
+				}
+			}
+		}
 
-	// No version set.
-	return false;
+		return $instance;
```

Comment:
This is _way_ better :)
If I may offer one more suggestion: instead of setting and carrying `$instance` - just return what you find, and return null here at the end?

If you do that, you should also use `empty( $instance )` instead of `! $instance` so it won't choke if `$instance` isn't uninitialized. 😉

Reply:
This whole foreach block should be replaced with a simply try/catch:

```php
try {
	return tribe( $class );
} catch ( RuntimeException $exception ) {
	return null;
}
```

Reply:
@sc0ttkclark did you try this? It didn't work for me, which is why I did it the way I did.

Reply:
This is supposed to work based on what I’ve read into tribe() code

Reply:
I am not so sure, but will have to test.
From the `tribe()` readme:
> @return mixed|object|Tribe__Container     The instance of the requested class. Please note that the cardinality of the class is controlled registering it as a singleton using `tribe_singleton` or `tribe_register`; if the `$slug_or_class` parameter is null then **the container itself will be returned**.

_Goes to test_
```
try {
	return tribe( 'foo.bar.baz' );
} catch ( RuntimeException $exception ) {
	return 'exception';
}
```

Returns `exception` - so it appears Scott is correct.